app/controllers/my/memberships_controller.rb

```diff
@@ -0,0 +1,20 @@
+class My::MembershipsController < My::ApplicationController
+  def destroy
+    membership.update left_at: current_time
+    current_user.update deactivated_at: current_time
+
+    sign_out
+
+    redirect_to root_path, notice: "Your membership to Ruby Australia has ceased."
+  end
+
+  private
+
+  def current_time
+    @current_time ||= Time.current
```

Comment:
🤔 ok to memoize this? does it snapshot for the request, or the lifecycle of the worker?

Also maybe `Time.zone.now`

Reply:
The choice to memoize was deliberate - so both `membership.left_at` and `current_user.deactivated_at` are exactly the same (which yes, is being a touch pedantic!)

As for `Time.zone.now` - my understanding is `Time.current` is the more modern approach - it uses `Time.zone.now` if a time-zone is set, otherwise `Time.now`. That said, not too fussed either way.